src/assets/static/pgp-worker.js

```diff
@@ -54,25 +56,21 @@ onmessage = function (event) {
         }
         postMessage({keys: decryptedPrivKeys, decryptPrivateKeys: true});
     } else if (event.data.decrypt) {
-        if (!event.data.content || !event.data.mailboxId) {
-            postMessage({decryptedContent: event.data.content, decrypted: true, callerId: event.data.callerId});
+        if (!event.data.mailboxId) {
+            postMessage({decryptedContent: {}, decrypted: true, callerId: event.data.callerId});
         } else {
-            decryptContent(event.data.content, decryptedPrivKeys[event.data.mailboxId])
-                .then((data) => {
-                    if (event.data.incomingHeaders) {
-                        decryptContent(event.data.incomingHeaders, decryptedPrivKeys[event.data.mailboxId])
-                            .then((incomingHeaders) => {
-                                postMessage({
-                                    incomingHeaders,
-                                    decryptedContent: data,
-                                    decrypted: true,
-                                    callerId: event.data.callerId
-                                });
-                            });
-                    } else {
-                        postMessage({decryptedContent: data, decrypted: true, callerId: event.data.callerId});
-                    }
+            decryptContent(event.data.mailData.content, decryptedPrivKeys[event.data.mailboxId]).then((content) => {
+                decryptContent(event.data.mailData.subject, decryptedPrivKeys[event.data.mailboxId]).then((subject) => {
+                    decryptContent(event.data.mailData.incomingHeaders, decryptedPrivKeys[event.data.mailboxId]).then((incomingHeaders) => {
```

Comment:
Are `incomingHeaders` always present in mail data now? The `if` condition has been removed.

Reply:
it has been handled inside decrypt content function rather than applying this condition for everyone separately.

Reply:
`function decryptContent(data, privKeyObj) {
    if (!data) {
        return Promise.resolve(data);
    }`

Reply:
ah, okay